pkg/compose/down.go

```diff
@@ -65,6 +65,21 @@ func (s *composeService) down(ctx context.Context, projectName string, options a
 		}
 	}
 
+	// Check requested services exists in model
+	var services []string
+	for _, service := range options.Services {
+		_, err := project.GetService(service)
+		if err != nil {
+			if options.Project != nil {
+				// ran with an explicit compose.yaml file, so we should not ignore
+				return err
+			}
+		} else {
+			services = append(services, service)
+		}
+	}
+	options.Services = services
```

Comment:
Can we extract this as a function, this way we won't need the `//nolint:gocyclo` and will make this code reusable elsewhere? WDYT?

Reply:
unfortunately, even with this extracted into a func, cyclomatic complexity still is 17

Reply:
moved to `checkSelectedServices`